EIPS/eip-6353.md

```diff
@@ -0,0 +1,333 @@
+
+Get all registered charity addresses.
+
+#### **getRate**
+
+Display for a user the rate of the default charity address that will receive donation.
+
+#### **setSpecificRate**
+
+Set personalized rate for charity address in {whitelistedRate}.
+
+| Parameter | Description |
+| ---------|-------------|
+| whitelistedAddr | The address to set as default. |
+| rate  | The personalised rate for donation. |
+
+#### **_returnRate**
+
+Return the rate to donate.
+
+| Parameter | Description |
+| ---------|-------------|
+| from | The address to get the rate of donation.
+
+#### **setSpecificDefaultAddress**
+
+Set for a user a default charity address that will receive donations. The default rate specified in {whitelistedRate} will be applied.
+
+| Parameter | Description |
+| ---------|-------------|
+| whitelistedAddr | The address to set as default.
+
+#### **setSpecificDefaultAddressAndRate**
+
+Set for a user a default charity address that will receive donations. The rate is specified by the user.
+
+| Parameter | Description |
+| ---------|-------------|
+| whitelistedAddr | The address to set as default. |
+| rate  | The personalized rate for donation.
+
+#### **specificDefaultAddress**
+
+Display for a user the default charity address that will receive donations. The default rate specified in {whitelistedRate} will be applied.
+
+#### **deleteDefaultAddress**
+
+Delete The Default Address and so deactivate donations.
+
+#### **charityInfo**
+
+Called with the charity address to determine if the contract whitelisted the address and if it is, the rate assigned.
+
+| Parameter | Description |
+| ---------|-------------|
+| addr | The Charity address queried for donnation information.
+
+## Rationale
+
+This standard provides functionality that allows token holders to donate easily. The donation when activated is done directly in the overridden `transfer`, `transferFrom`, and `approve` functions.
+
+The donation is a percentage-based rate model, but the calculation can be done differently. Also, donations could be in other functions than the `transfer` function.
+
+ To manage the whitelist the owner of the contract can choose to whitelist charity addresses by using an array and keeping track of the "active" status with a mapping. The donation address can also be a single address chosen by the owner of the contract and modified by period.
+
+ They can also choose to store the donation in the contract or in another and add a withdrawal or claimable function so the charity can claim the allocated amount of token themselves, so the transfer function will be triggered by the charity and not the token holder.
+
+## Backwards Compatibility
+
+There are no backward compatibility issues, this implementation is an extension of the functionality of [EIP-20](./eip-20.md). This EIP is fully backward compatible and introduces new functionality retaining the core interfaces and functionality of the [EIP-20](./eip-20.md) standard.
```

Comment:
There is a small backwards compatibility issue. Without EIP-6353, if an account has sufficient balance, a `transfer` will succeed. With EIP-6353, it's possible for such a transfer to fail.